core/src/main/java/bisq/core/dispute/DisputeResolverService.java

```diff
@@ -0,0 +1,127 @@
+                                   ErrorMessageHandler errorMessageHandler) {
+        log.debug("addDisputeResolver disputeResolver.hashCode() " + disputeResolver.hashCode());
+        if (!BisqEnvironment.getBaseCurrencyNetwork().isMainnet() ||
+                !Utilities.encodeToHex(disputeResolver.getRegistrationPubKey()).equals(DevEnv.DEV_PRIVILEGE_PUB_KEY)) {
+            boolean result = p2PService.addProtectedStorageEntry(disputeResolver, true);
+            if (result) {
+                log.trace("Add disputeResolver to network was successful. DisputeResolver.hashCode() = " + disputeResolver.hashCode());
+                resultHandler.handleResult();
+            } else {
+                errorMessageHandler.handleErrorMessage("Add disputeResolver failed");
+            }
+        } else {
+            log.error("Attempt to publish dev disputeResolver on mainnet.");
+            errorMessageHandler.handleErrorMessage("Add disputeResolver failed. Attempt to publish dev disputeResolver on mainnet.");
+        }
+    }
+
+    public void removeDisputeResolver(T disputeResolver,
+                                      ResultHandler resultHandler,
+                                      ErrorMessageHandler errorMessageHandler) {
+        log.debug("removeDisputeResolver disputeResolver.hashCode() " + disputeResolver.hashCode());
+        if (p2PService.removeData(disputeResolver, true)) {
+            log.trace("Remove disputeResolver from network was successful. DisputeResolver.hashCode() = " + disputeResolver.hashCode());
+            resultHandler.handleResult();
+        } else {
+            errorMessageHandler.handleErrorMessage("Remove disputeResolver failed");
+        }
+    }
+
+    public P2PService getP2PService() {
+        return p2PService;
+    }
+
+    public Map<NodeAddress, T> getDisputeResolvers() {
+        final List<String> bannedDisputeResolvers;
+        if (filterManager.getFilter() != null) {
+            bannedDisputeResolvers = getDisputeResolversFromFilter();
+        } else {
+            bannedDisputeResolvers = null;
+        }
+        if (bannedDisputeResolvers != null)
+            log.warn("bannedDisputeResolvers=" + bannedDisputeResolvers);
+        Set<T> disputeResolverSet = getCollect(bannedDisputeResolvers);
+
+        Map<NodeAddress, T> map = new HashMap<>();
+        for (T disputeResolver : disputeResolverSet) {
+            NodeAddress disputeResolverNodeAddress = disputeResolver.getNodeAddress();
+            if (!map.containsKey(disputeResolverNodeAddress))
+                map.put(disputeResolverNodeAddress, disputeResolver);
+            else
+                log.warn("disputeResolverAddress already exist in disputeResolver map. Seems an disputeResolver object is already registered with the same address.");
+        }
+        return map;
+    }
+
+    @NotNull
+    protected abstract Set<T> getCollect(List<String> bannedDisputeResolvers);
```

Comment:
`getCollect()` is not a great name, `getSet()` might be better

Reply:
Upps, was not intended. Will change it.

Reply:
Upps, was not intended. Will change it.